### DISCUSSION TITLE
Examples with explanation

### DISCUSSION THREAD

**Discussion by babaniyi** [StatMixedML/XGBoostLSS#44] [Ideas]
This is a beautiful project and should be more popular. However, the example notebooks are challenging to understand for a first-time user. It's difficult to understand the task, goal, or aim of each of the examples and one only sees plots and code snippets, I believe an explanation or description would go a long way in the adoption of this package.

I would be glad if someone would propose an example where XGBoostLSS is applied to open-sourced data (not simulated, say Kaggle) to see how it works.

**Comment by StatMixedML** [comment]
Thanks @babaniyi for your interest in the project.

I have added a [documentation](https://statmixedml.github.io/XGBoostLSS/) and explained the basic usage in [Gaussian_Regression](https://statmixedml.github.io/XGBoostLSS/examples/Gaussian_Regression/)

**Comment by StatMixedML** [comment]
@babaniyi Do the examples help in getting more familiar with the package?

**Reply by babaniyi** [reply]
Thanks for taking the time to work on the latest release, it is a vast improvement. Also, I am happy about the [California data example](https://statmixedml.github.io/XGBoostLSS/examples/Gamma_Regression_CaliforniaHousing/) used, it makes it to have a real-world application touch. In September, I aim to create a pull request for an example where I will write a detailed explanation and how-to-use case.
By the way, I have been thinking about applications for massive datasets (not in pandas) say stored as a spark data frame, do you have any plans for scaling in the future?

Once again, thanks for working tirelessly on this package.

**Comment by StatMixedML** [comment]
> In September, I aim to create a pull request for an example where I will write a detailed explanation and how-to-use case.

Great, looking forward!

> By the way, I have been thinking about applications for massive datasets (not in pandas) say stored as a spark data frame

Native XGBoost supports distributed computation via PySpark, Dask and Ray. I haven't tested XGBoostLSS on these. Definitely a good future extension!
